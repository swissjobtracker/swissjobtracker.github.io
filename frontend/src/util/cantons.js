export default [
  'ag',
  'ai',
  'ar',
  'be',
  'bl',
  'bs',
  'fr',
  'ge',
  'gl',
  'gr',
  'ju',
  'lu',
  'ne',
  'nw',
  'ow',
  'sg',
  'sh',
  'so',
  'sz',
  'tg',
  'ti',
  'ur',
  'vd',
  'vs',
  'zg',
  'zh'
].sort()
