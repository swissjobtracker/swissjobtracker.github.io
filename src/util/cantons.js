export default [
  {
    label: 'Aargau',
    value: 'ag',
  },
  {
    label: 'Appenzell Innerrhoden',
    value: 'ai',
  },
  {
    label: 'Appenzell Ausserrhoden',
    value: 'ar',
  },
  {
    label: 'Bern',
    value: 'be',
  },
  {
    label: 'Basel-Landschaft',
    value: 'bl',
  },
  {
    label: 'Basel-Stadt',
    value: 'bs',
  },
  {
    label: 'Fribourg',
    value: 'fr',
  },
  {
    label: 'Genève',
    value: 'ge',
  },
  {
    label: 'Glarus',
    value: 'gl',
  },
  {
    label: 'Grishun',
    value: 'gr',
  },
  {
    label: 'Jura',
    value: 'ju',
  },
  {
    label: 'Luzern',
    value: 'lu',
  },
  {
    label: 'Neuchâtel',
    value: 'ne',
  },
  {
    label: 'Nidwalden',
    value: 'nw',
  },
  {
    label: 'Obwalden',
    value: 'ow',
  },
  {
    label: 'St. Gallen',
    value: 'sg',
  },
  {
    label: 'Schaffhausen',
    value: 'sh',
  },
  {
    label: 'Solothurn',
    value: 'so',
  },
  {
    label: 'Schwyz',
    value: 'sz',
  },
  {
    label: 'Thurgau',
    value: 'tg',
  },
  {
    label: 'Ticino',
    value: 'ti',
  },
  {
    label: 'Uri',
    value: 'ur',
  },
  {
    label: 'Vaud',
    value: 'vd',
  },
  {
    label: 'Vallais',
    value: 'vs',
  },
  {
    label: 'Zug',
    value: 'zg',
  },
  {
    label: 'Zürich',
    value: 'zh'
  }
].sort((a, b) => a.value > b.value ? 1 : -1)
