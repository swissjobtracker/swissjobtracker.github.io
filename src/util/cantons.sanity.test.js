import {cantons, getCantonByCode} from './cantons'

test('cantons definition is intact', () => {
  expect(cantons).toMatchSnapshot()
})

test('getCantonByCode returns canton labels', () => {
  expect(getCantonByCode('TI')).toEqual('Ticino')
})

test('getCantonByCode returns undefined when not found', () => {
  expect(getCantonByCode('xx')).toBeUndefined()
})
