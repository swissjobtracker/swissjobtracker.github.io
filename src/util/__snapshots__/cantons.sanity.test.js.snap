// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cantons definition is intact 1`] = `
Array [
  Object {
    "label": "Aargau",
    "value": "ag",
  },
  Object {
    "label": "Appenzell Innerrhoden",
    "value": "ai",
  },
  Object {
    "label": "Appenzell Ausserrhoden",
    "value": "ar",
  },
  Object {
    "label": "Bern",
    "value": "be",
  },
  Object {
    "label": "Basel-Landschaft",
    "value": "bl",
  },
  Object {
    "label": "Basel-Stadt",
    "value": "bs",
  },
  Object {
    "label": "Fribourg",
    "value": "fr",
  },
  Object {
    "label": "Genève",
    "value": "ge",
  },
  Object {
    "label": "Glarus",
    "value": "gl",
  },
  Object {
    "label": "Grishun",
    "value": "gr",
  },
  Object {
    "label": "Jura",
    "value": "ju",
  },
  Object {
    "label": "Luzern",
    "value": "lu",
  },
  Object {
    "label": "Neuchâtel",
    "value": "ne",
  },
  Object {
    "label": "Nidwalden",
    "value": "nw",
  },
  Object {
    "label": "Obwalden",
    "value": "ow",
  },
  Object {
    "label": "St. Gallen",
    "value": "sg",
  },
  Object {
    "label": "Schaffhausen",
    "value": "sh",
  },
  Object {
    "label": "Solothurn",
    "value": "so",
  },
  Object {
    "label": "Schwyz",
    "value": "sz",
  },
  Object {
    "label": "Thurgau",
    "value": "tg",
  },
  Object {
    "label": "Ticino",
    "value": "ti",
  },
  Object {
    "label": "Uri",
    "value": "ur",
  },
  Object {
    "label": "Vaud",
    "value": "vd",
  },
  Object {
    "label": "Vallais",
    "value": "vs",
  },
  Object {
    "label": "Zug",
    "value": "zg",
  },
  Object {
    "label": "Zürich",
    "value": "zh",
  },
]
`;
